R/bench-summ.R

```diff
@@ -0,0 +1,166 @@
+    mutate(regulon_time = sum(statistic_time)) %>%
+    select(set_name, bench_name, statistic,
+           filter_crit, statistic_time, regulon_time)
+
+  # Join AUROC, PRAUC, Coverage, and Comp time
+  summary_table <- auroc_tibble %>%
+    inner_join(prauc_tibble %>%
+                 rename(pr_auc = auc),
+               by = c("set_name", "bench_name", "statistic", "filter_crit")) %>%
+    distinct() %>%
+    inner_join(x=.,
+               y=(roc %>%
+                    group_by(name_lvl) %>%
+                    summarise(source_cov = coverage,
+                              condition_cov = n,
+                              roc_neg=tn) %>%
+                    distinct() %>%
+                    ungroup() %>%
+                    separate(col="name_lvl",
+                             into=c("set_name", "bench_name", "filter_crit"),
+                             sep="\\.")),
+               by = c("set_name", "bench_name", "filter_crit")) %>%
+    distinct() %>%
+    inner_join(x=.,
+               y=(pr %>%
+                    group_by(name_lvl) %>%
+                    summarise(
+                              pr_neg=tn) %>%
+                    distinct() %>%
+                    ungroup() %>%
+                    separate(col="name_lvl",
+                             into=c("set_name", "bench_name", "filter_crit"),
+                             sep="\\.")),
+               by = c("set_name", "bench_name", "filter_crit")) %>%
+    distinct() %>%
+    inner_join(x=.,
+               y=comp_time,
+               by = c("set_name", "bench_name", "filter_crit", "statistic")) %>%
+    distinct()
+
+  bench_summary <- list(summary_table, roc_plot, pr_plot,
+                        auroc_plot, auroc_heat, pr_heat)
+
+  names(bench_summary) <- c("summary_table", "roc_plot", "pr_plot",
+                            "auroc_plot", "auroc_heat", "pr_heat")
+
+  return(bench_summary)
+}
+
+
+
+#' Helper function to format (PR) Receiver Operator Curve results
+#' @param .res_tibble formatted bench result tibble with added auroc column
+#' @param roc_column PR/ROC column to format
+#' @return returns
+format_roc <- function(.res_tibble, roc_column){
+  apply(.res_tibble, 1, function(df) {
+    df[roc_column] %>%
+      enframe() %>%
+      as_tibble() %>%
+      unnest(value) %>%
+      mutate(set_name = df$set_name,
+             bench_name = df$bench_name,
+             filter_crit = df$filter_crit,
+             statistic = df$statistic) %>%
+      unite("name_lvl", set_name, bench_name,
+            filter_crit, remove = F, sep = ".") %>%
+      unite("run_key", set_name, bench_name,
+            statistic, filter_crit, remove = F)
+  }) %>%
+    do.call(rbind, .)
+}
+
+
+#' Helper function to produce AUROC heatmap
+#' @param auroc_tibble Tibble with calculated AUROC
+#' @return returns an AUROC or Precision-Recall AUC heatmap
+#' @import pheatmap ggplot2
+get_auroc_heat <- function(auroc_tibble){
```

Comment:
Since the values go between zero and one, it would be convenient to be able to decide if scale of the legend should or should not be fixed.

Reply:
IMHO, this is maybe beyond the purpose of the package. It should be up to the user to change them if they want them to be of publication quality.